articles/introduction-to-dynamic-programming/index.md

```diff
@@ -0,0 +1,147 @@
+In the previous article on recursion, we explored how we can break a problem into smaller sub-problems and solve them individually. However, recursion is not the most optimal technique and has its share of obstacles. Fortunately, there is a powerful algorithmic technique called dynamic programming that helps us overcome the hurdles posed by recursion and solve problems optimally. In this article, we will understand how dynamic programming (popularly referred to as DP) works by solving coding questions.
+
+### Table of contents
+[Prerequisites](#prerequisites)
+
+[Understanding DP](#understanding-dp)
+
+[Fibonacci Numbers](#fibonacci-numbers)
+
+[Conclusion](#conclusion)
+
+### Prerequisites
+1. Basic understanding of [Programming](https://www.tutorialspoint.com/computer_programming/computer_programming_basics.htm) and [time complexities](https://www.freecodecamp.org/news/time-complexity-of-algorithms/)
+
+2. Basic knowledge of [Python](https://www.python.org/about/gettingstarted/)
+
+3. A basic understanding of [Recursion](https://www.section.io/engineering-education/introduction-to-recursion)
+
+### Understanding DP
+According to [educative](https://www.educative.io/courses/grokking-dynamic-programming-patterns-for-coding-interviews/m2G1pAq0OO0), Dynamic Programming (DP) is an algorithmic technique for solving an optimization problem by breaking it down into simpler subproblems and utilizing the fact that the optimal solution to the overall problem depends upon the optimal solution to its subproblems. It sounds similar to recursion. That's because DP is just an optimized version of recursion. Before we delve into DP, let us look at the drawbacks of recursion:
```

Comment:
Did educative define DP? It would be better if the definition is stated and the link is inserted without saying according to educative.